Simulator/Recorders/Hdf5Recorder.h

```diff
@@ -127,31 +129,49 @@ class Hdf5Recorder : public Recorder {
          if (dataType_ == typeid(uint64_t).name()) {
             hdf5Datatype_ = PredType::NATIVE_UINT64;
          } else if (dataType_ == typeid(bool).name()) {
-            hdf5Datatype_ = PredType::NATIVE_UINT8;
+            hdf5Datatype_ = PredType::NATIVE_INT;
          } else if (dataType_ == typeid(int).name()) {
             hdf5Datatype_ = PredType::NATIVE_INT;
-         } else if (dataType_ == typeid(float).name()) {
+         } else if (dataType_ == typeid(BGFLOAT).name()) {
```

Comment:
Actually, I think the original here (comparing to `float`) is better. `BGFLOAT` will either be a `float` or a `double`, and this will then work appropriately either way.

Reply:
I will change it back